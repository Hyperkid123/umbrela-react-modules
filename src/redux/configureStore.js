import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
