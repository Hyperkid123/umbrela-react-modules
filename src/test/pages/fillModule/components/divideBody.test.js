import React from 'react'
import DivideBody, {DivideBody as Snapshot} from '../../../../pages/fillModule/components/divideBody';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Divide body component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.answer = null;
    initialProps.translate = jest.fn();
    initialProps.startResearch = jest.fn();
    initialProps.question = {
      questionId: 0,
      scalePoints: 100,
      options: [],
    }
    initialProps.answerDivideQuestion = jest.fn();
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  const mockStore = configureStore();
  let initialState = {};
  let store;
  beforeEach(() => {
    initialState = {
      answers: {
        "0": {
          options: []
        }
      },
      locale: {
        languages: [{
          name: "English",
          code: 'en',
          active: true,
        }],
        translations: {}
      }
    }
  });

  it('should create component correctly', () => {
    store = mockStore(initialState);
    let wrapper = shallow(
      <DivideBody
        question={{questionId: 0, scalePoints: 100, options: []}}
        store={store}
      />);
    expect(wrapper).toBeDefined();
  });

  it('should reder with options', () => {
    const question = {
      options: [{
        optionId: 0,
        optionTitle: 'option 1'
      }, {
        optionId: 1,
        optionTitle: 'option 2'
      }]
    }
    const tree = shallow(
      <Snapshot {...initialProps} question={question}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should add scale points to option', () => {
    const options = [{}, {}]
    const answerDivideQuestion = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} answerDivideQuestion={answerDivideQuestion}/>
    )
    wrapper.instance().handleDivideChange(0, {target: {value: 50}}, options.length, 0, 1);
    expect(answerDivideQuestion).toHaveBeenCalledWith(0, 50, 2, 0);
  })

  it('should sub answer, not enough points', () => {
    const options = [{
      optionId: 0
    }, {
      optionId: 1
    }]
    const answer = [50, 50]
    const question = {
      scalePoints: 100,
      options,
    }
    const answerDivideQuestion = jest.fn();
    const wrapper = shallow(
      <Snapshot
        {...initialProps}
        question={question}
        answerDivideQuestion={answerDivideQuestion}
        answer={answer}
      />
    )
    wrapper.instance().handleDivideChange(0, {target: {value: 3}}, options.length, 0, 100);
    expect(answerDivideQuestion).toHaveBeenCalledWith(0, 3, 2, 0);
  })

  it('should not update answer, not enough points', () => {
    const options = [{
      optionId: 0
    }, {
      optionId: 1
    }]
    const answer = [50, 50]
    const question = {
      scalePoints: 0,
      options,
    }
    const answerDivideQuestion = jest.fn();
    const wrapper = shallow(
      <Snapshot
        {...initialProps}
        question={question}
        answerDivideQuestion={answerDivideQuestion}
        answer={answer}
      />
    )
    wrapper.instance().handleDivideChange(0, {target: {value: 50}}, options.length, 0, 100);
    expect(answerDivideQuestion).toHaveBeenCalledTimes(0);
  })
});
