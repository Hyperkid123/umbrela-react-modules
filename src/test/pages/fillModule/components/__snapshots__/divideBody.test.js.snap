// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divide body component should reder with options 1`] = `
<styled.div
  column={true}
>
  <label>
    : 
    NaN
  </label>
  <styled.div
    column={true}
  >
    <styled.div
      key="0"
    >
      <TextField
        min={0}
        name="divide_input_0"
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        select={false}
        style={
          Object {
            "alignSelf": "flex-end",
            "flexShrink": 0,
            "width": 150,
          }
        }
        type="number"
        value={0}
      />
      <styled.div
        column={true}
        grow={true}
        style={
          Object {
            "marginLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        option 1
        <WithStyles(LinearProgress)
          mode="determinate"
          value={NaN}
        />
      </styled.div>
    </styled.div>
    <styled.div
      key="1"
    >
      <TextField
        min={0}
        name="divide_input_1"
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        select={false}
        style={
          Object {
            "alignSelf": "flex-end",
            "flexShrink": 0,
            "width": 150,
          }
        }
        type="number"
        value={0}
      />
      <styled.div
        column={true}
        grow={true}
        style={
          Object {
            "marginLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        option 2
        <WithStyles(LinearProgress)
          mode="determinate"
          value={NaN}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Divide body component should render correctly empty 1`] = `
<styled.div
  column={true}
>
  <label>
    : 
    100
  </label>
  <styled.div
    column={true}
  />
</styled.div>
`;
